cupy/core/core.pyx

```diff
@@ -422,8 +422,13 @@ cdef class ndarray:
         finally:
             runtime.setDevice(dev_id)
         newarray = ndarray(x.shape, dtype=x.dtype)
+        if not x._c_contiguous and not x._f_contiguous:
+            raise NotImplementedError(
+                'CuPy cannot copy non-contiguous array between devices.')
```

Comment:
Just to make sure, do you mean it's a "TODO", or actually that it's never going to be supported (in which case we should not use `NotImplementedError`)?

Reply:
It means TODO. In some cases, we can implement it.